src/ui/TextEditor.tsx

```diff
@@ -9,45 +7,37 @@ import SHARED from "../shared";
 import { API } from "./ToggleEditor";
 import { AST } from "../ast";
 import { Editor } from "codemirror";
-import { AppDispatch } from "../store";
 
 // CodeMirror APIs that we need to disallow
 // NOTE(Emmanuel): we should probably block 'on' and 'off'...
 const unsupportedAPIs = ["startOperation", "endOperation", "operation"];
 
-const mapStateToProps = () => ({});
-const mapDispatchToProps = (dispatch: AppDispatch) => ({
-  dispatch,
-  setAnnouncer: (announcer: HTMLElement) =>
-    dispatch({ type: "SET_ANNOUNCER", announcer }),
-});
-const connector = connect(mapStateToProps, mapDispatchToProps);
-
-type Props = ConnectedProps<typeof connector> & {
+type Props = {
   cmOptions?: {};
   value: string;
   onBeforeChange?: IUnControlledCodeMirror["onBeforeChange"];
   onMount: (ed: Editor, api: API, ast: AST) => void;
-  setAnnouncer: Function;
   api?: API;
   passedAST?: AST;
 };
 
-class TextEditor extends Component<Props> {
+const TextEditor = (props: Props) => {
+  const { onMount, passedAST, value, onBeforeChange, cmOptions } = props;
+
   /**
    * @internal
```

Comment:
These @internal comments are no longer necessary because the scope by which they can be accessed is now limited to the function. hurray!